src/store/modules/tester.js

```diff
@@ -1,369 +1,288 @@
+        { label: '\u2192', x: 17.25, y: 5.5 },
+        { label: '0', x: 18.5, y: 5.5, w: 2 },
+        { label: '.', x: 20.5, y: 5.5 }
+      ]
+    }
   }
 };
 
 const getters = {};
 
+function mapKeymap(store, arr) {
+  return arr.map(code => {
+    const meta = store.getters['keycodes/lookupKeycode'](code);
+    return {
+      ...meta
+    };
+  });
+}
+
 const actions = {
   init({ state, commit }) {
     const store = this;
     commit('setKeymap', [
-      [
-        'KC_ESC',
-        'KC_F1',
-        'KC_F2',
-        'KC_F3',
-        'KC_F4',
-        'KC_F5',
-        'KC_F6',
-        'KC_F7',
-        'KC_F8',
-        'KC_F9',
-        'KC_F10',
-        'KC_F11',
-        'KC_F12',
-        'KC_PSCR',
-        'KC_SLCK',
-        'KC_PAUS',
-        'KC_GRV',
-        'KC_1',
-        'KC_2',
-        'KC_3',
-        'KC_4',
-        'KC_5',
-        'KC_6',
-        'KC_7',
-        'KC_8',
-        'KC_9',
-        'KC_0',
-        'KC_MINS',
-        'KC_EQL',
-        'KC_BSPC',
-        'KC_INS',
-        'KC_HOME',
-        'KC_PGUP',
-        'KC_NLCK',
-        'KC_PSLS',
-        'KC_PAST',
-        'KC_PMNS',
-        'KC_TAB',
-        'KC_Q',
-        'KC_W',
-        'KC_E',
-        'KC_R',
-        'KC_T',
-        'KC_Y',
-        'KC_U',
-        'KC_I',
-        'KC_O',
-        'KC_P',
-        'KC_LBRC',
-        'KC_RBRC',
-        'KC_BSLS',
-        'KC_DEL',
-        'KC_END',
-        'KC_PGDN',
-        'KC_P7',
-        'KC_P8',
-        'KC_P9',
-        'KC_PPLS',
-        'KC_CAPS',
-        'KC_A',
-        'KC_S',
-        'KC_D',
-        'KC_F',
-        'KC_G',
-        'KC_H',
-        'KC_J',
-        'KC_K',
-        'KC_L',
-        'KC_SCLN',
-        'KC_QUOT',
-        'KC_ENT',
-        'KC_P4',
-        'KC_P5',
-        'KC_P6',
-        'KC_LSFT',
-        'KC_Z',
-        'KC_X',
-        'KC_C',
-        'KC_V',
-        'KC_B',
-        'KC_N',
-        'KC_M',
-        'KC_COMM',
-        'KC_DOT',
-        'KC_SLSH',
-        'KC_RSFT',
-        'KC_UP',
-        'KC_P1',
-        'KC_P2',
-        'KC_P3',
-        'KC_PENT',
-        'KC_LCTL',
-        'KC_LGUI',
-        'KC_LALT',
-        'KC_SPC',
-        'KC_RALT',
-        'KC_RGUI',
-        'KC_APP',
-        'KC_RCTL',
-        'KC_LEFT',
-        'KC_DOWN',
-        'KC_RGHT',
-        'KC_P0',
-        'KC_PDOT'
-      ].map(code => {
-        const meta = store.getters['keycodes/lookupKeycode'](code);
-        return {
-          ...meta
-        };
-      })
+      mapKeymap(store, ANSIkeymap),
+      mapKeymap(store, ISOkeymap)
     ]);
     return state.keymap;
   }
 };
 
 const mutations = {
+  setLayout(state, layout) {
+    state.layout = layout;
+  },
   setKeymap(state, keymap) {
     state.keymap = keymap;
   },
-  setActive(state, pos) {
-    Vue.set(state.keymap[0][pos], 'active', true);
-    Vue.set(state.keymap[0][pos], 'detected', false);
+  setActive(state, { keymapIdx, pos }) {
```

Comment:
I don't like this. It's too messy. We should hide the layout abstraction inside the store, not the component. The store should know what layout is currently selected and provide getters. This just exposes too much inner implementation.